### PR TITLE
resolve show and season on season and episode

### PR DESCRIPTION
**Pull request by fredrikved**
branch for implementing show -> seasons -> episodes and episode -> season -> show

also added a view here for expanded items, to avoid building duplicate queries (single item loading and list loading (children))

### CODE REVIEW

File: backend/graph/model/episode.go [hunk 1]
```diff
@@ -23,13 +24,25 @@ func EpisodeFromSQL(ctx context.Context, row *sqlc.EpisodeExpanded) *Episode {
...
+	}
 
 	episode := &Episode{
 		Chapters:         []*Chapter{}, // Currently not supported
 		ID:               fmt.Sprintf("%d", row.ID),
 		Title:            titleMap.Get(languages),
 		Description:      descriptionMap.Get(languages),
-		ExtraDescription: extraDescriptionMap.Get(languages),
+		ExtraDescription: extraDescription,
+		Season:           season,
```
Comment (KillerX) [inline backend/graph/model/episode.go:45]: Does the resolver still run if you assign the object here?

File: backend/graph/resolver.go [hunk 2]
```diff
@@ -63,3 +67,23 @@ func resolverForIntID[t restrictedItem, r any](ctx context.Context, id string, l
 
 	return resolverFor(ctx, int(intID), loader, converter)
 }
+
+func itemsResolverFor[k comparable, t restrictedItem, r any](ctx context.Context, id k, loader *dataloader.Loader[k, []*t], converter func(context.Context, *t) *r) ([]*r, error) {
```
Comment (KillerX) [inline backend/graph/resolver.go:71]: The signatures are starting to resemble C++ :D

**Review by KillerX — APPROVED** [review]
I assume you have checked what I wrote in the other comment. Looks nice!

File: backend/graph/model/episode.go (hunk 1 above)
Comment (fredrikved) [reply backend/graph/model/episode.go:45]: Yes. Runs as expected, but I could avoid assigning it and instead add SeasonID to Episode, etc, as I only need the ID for the resolvers to work.

Comment (KillerX) [reply backend/graph/model/episode.go:45]: I think this is perfectly fine then.

**Pull request merged**